File: Practicas/RobocodeLabo-1.10.0/RobocodeLabo/src/estrategias/Estrategia.java
package RobotGod.estrategias;

import NicoustinRobot;
import robocode.JuniorRobot;
import robocode.WinEvent;

public interface Estrategia {

    void runB(NicoustinRobot robot);
    void onScannedRobot();
    void onHitByBullet();
    void onHitWall();
    void setRobot(NicoustinRobot robot);
    void onWin(WinEvent e);
    
    // Nuevo método para evaluación de estrategia
    String evalStrat();
}

File: Practicas/Robocode-Kotlin-2/src/estrategias/Estrategia.kt
package estrategias

import laboratorio.NicoustinRobot
import robocode.WinEvent

interface Estrategia {
    fun runB(robot: NicoustinRobot)
    fun onScannedRobot()
    fun onHitByBullet()
    fun onHitWall()
    fun setRobot(robot: NicoustinRobot)
    fun onWin(event: WinEvent)
}